### ISSUE TITLE
KeyError: '_errors' during highstate run

### ISSUE THREAD

**Issue by babilen** [saltstack/salt#46775]
### Description of Issue/Question

We are seeing the following error one *some* of our minions during a `state.apply` on all of them. Their configuration should be identical. It is also non-deterministic which minions are affected by this, in that the set of affected minions is different each time it happens.

```
minion_id:
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1482, in _thread_return
        return_data = executor.execute()
      File "/usr/lib/python2.7/dist-packages/salt/executors/direct_call.py", line 28, in execute
        return self.func(*self.args, **self.kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/state.py", line 601, in apply_
        return sls(mods, **kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/state.py", line 1081, in sls
        err += __pillar__['_errors']
      File "/usr/lib/python2.7/dist-packages/salt/utils/context.py", line 211, in __getitem__
        return self._dict()[key]
    KeyError: '_errors'
```

### Versions Report

Master:
```
$ salt --versions-report
Salt Version:
           Salt: 2016.11.9
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Nov 24 2017, 17:33:09)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1
 
System Versions:
           dist: debian 9.3 
        machine: x86_64
        release: 4.9.0-5-amd64
         system: Linux
        version: debian 9.3
```

Minion:
```
   Salt Version:
               Salt: 2016.11.9
     
    Dependency Versions:
               cffi: 1.9.1
           cherrypy: Not Installed
           dateutil: 2.5.3
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.9.4
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.8
       mysql-python: Not Installed
          pycparser: 2.17
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.13 (default, Nov 24 2017, 17:33:09)
       python-gnupg: Not Installed
             PyYAML: 3.12
              PyZMQ: 16.0.2
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.4.3
                ZMQ: 4.2.1
     
    System Versions:
               dist: debian 9.4 
            machine: x86_64
            release: 4.9.0-6-amd64
             system: Linux
            version: debian 9.4
```
I have, however, seen it in later (i.e. `2017.7.4`) versions also.

I would appreciate any help in debugging this problem as it makes our highstate runs non-deterministic as it won't execute correctly on all targeted minions.

**Comment by garethgreenaway** [comment]
@babilen 2016.11 .x is only receiving CVE related fixes, not bug fixes, so we would fix this issue against 2017.7.x and higher.  Can you provide the errors that you're seeing on those versions?  Thanks!

**Comment by babilen** [comment]
Upgraded everything to `2017.7.4` and can not reproduce the problem right now. I shall re-open this issue if I run into it again.
